### ISSUE TITLE
no-invalid-this false flags usage in ES2015 static methods

### ISSUE THREAD

**Issue by aparajita** [eslint/eslint#4669]
**What version are you using?**

1.10.3

**What did you do?**

Use `this` in a static class method, like this:

```js
'use strict';

/* eslint-env node, es6 */
/* eslint no-invalid-this: 2 */

class Foo
{
    static foo()
    {
        console.log('foo');
        this.bar();
    }

    static bar()
    {
        console.log('bar');
    }
}

const f = new Foo();
f.constructor.foo();

// console output
foo
bar
```

**What happened?**

When I run `eslint --no-eslintrc` on this code, I get this error:

```text
  11:9  error  Unexpected `this`  no-invalid-this
```

**What did you expect to happen?**

There shouldn't be an error, this code works fine. Using `this` in a static function to refer to the class is valid in node, Safari and Chrome.

**Comment by eslintbot** [comment]
@aparajita Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by ilyavolodin** [comment]
ESLint doesn't really support experimental features. You might need to make an issue for https://github.com/babel/eslint-plugin-babel

**Comment by nzakas** [comment]
@ilyavolodin this is ES6, not experimental. 

This is a bug.

**Comment by ilyavolodin** [comment]
Ahh.. sorry about that. I was under the impression that `static` was not part of ES6 spec.

**Comment by aparajita** [comment]
It's actually a wrong design decision. The docs for that rule explicitly show this use case as being invalid.

**Comment by aparajita** [comment]
Thanks!
